### PR TITLE
Update diagnostics docs

### LINKED ISSUE #945: Include examples of other diagnostics
At http://dask.pydata.org/en/latest/diagnostics.html we have an example showing the `Profiler`.  It would be helpful to have visual examples of the `CacheProfiler` and `ResourceProfiler` as well.

### PR DESCRIPTION
**Pull request by jcrist**
This updates the diagnostics documentation to cover all three profilers, as well as well as cleans up a few errors in the docstrings. I added a prose section covering using all 3 profilers to evaluate the performance of all some `dask.array` code.

There were also a few cleanups to formatting of the bokeh plots, mostly to the layout.

Fixes #945.

### CODE REVIEW

File: docs/source/diagnostics.rst [hunk 1]
```diff
@@ -67,62 +106,157 @@ During execution the profiler records the following information for each task:
...
+
+
+.. raw:: html
+
+    <iframe src="_static/profile.html"
+            marginwidth="0" marginheight="0" scrolling="no"
+            width="650" height="330" style="border:none"></iframe>
+
+To view multiple profilers at the same time, the ``dask.diagnostics.visualize``
+function can be used. This takes a list of profilers, and creates a verticle
```
Comment (mrocklin) [inline docs/source/diagnostics.rst]: typo: verticle -> vertical

**Comment by mrocklin** [comment]
This is a great read.  +1 except for the typo

File: docs/source/diagnostics.rst (hunk 1 above)
Comment (jcrist) [inline docs/source/diagnostics.rst]: Fixed. Thanks.

**Comment by jcrist** [comment]
Merging.

**Pull request merged**